### ISSUE TITLE
Add support for spars

### ISSUE THREAD

**Issue by jasonhamilton** [jasonhamilton/hotwing-core#1]

**Comment by monkeypants** [comment]
some questions/thoughts about this:
* how would spars look in the config file? `[spar-1]`, ..., `[spar-n]` 
* how to define spar geometry? types like rectangular, ellipse, round, half-round, etc?
* how to define spar orientation? vertical, normal to wing surface, some degrees from vertical?
* would each spar have an inboard and outboard shape (for e.g. tapered spars)?
* would each spar have an inboard and outboard position? absolute position or relative to wing surface?
* would control surfaces be segmented off by defining a "spar", or should there be separate syntax for that?
* for spars that are not adjacent to airfoil surface, would you specify the path into the spar or assume it comes vertically down to the centre of the spar, or some other path.

**Comment by monkeypants** [comment]
Taking my first look at the code, with spar support in mind. Would the place to start be the Rib class (consider the signature, how to pass in spars?) and test it so it renders OK with the various types of spar that seem like good ideas, then work out to panels/cutting strategies (in hotwing-cli)?

**Comment by jasonhamilton** [comment]
Excellent ideas! It’s been awhile since I’ve looked at the code but I think you would want the Panel class to handle the spars.  The rib is just a cross section of a wing, whereas the Panel is more like a solid wing.

My initial thought would be to create a Spar class that’s passed to the Panel class?  I’m happy to take a shot at a simple Spar implementation unless you want go have a go at it.  

It’s been a few years since I’ve cut a wing so I don’t have my hot wire set up to test it.  I can test in a sim, however.

**Comment by monkeypants** [comment]
Yes a Spar class makes sense. Spars are lofted shapes (like Panels), oriented in a frame relative to Panels. They need to share the same ends as the Panel if your hot-wire machine (or sim) is anything like mine :). Actually like you I'm currently without a hot-wire cutter (my machine is on loan and might not come back), I discovered your code because I was thinking about making another one... would you just use LinuxCNC to simulate/test? I think that's what I would do too. FWIW the LinuxCNC config of my old machine is [here](https://github.com/monkeypants/foamcutter).

Maybe I'm over-thinking this bit, but in real life a Spar is more like a Wing than a Panel (i.e. for multi-panel wings, most spars will be beams that span multiple panels). So the thing that shares the ends of the Panel is a SparSegment (and a Panel is a WingSegment). Now that pedantry is out of my system I'm happy to call them Spars and add them to Panels.

The Spars I can imagine finding useful are:
* non-tapered round tubes that are embedded in the wing, often not touching either surface, and accessed via a slot from above or below.
* full height sections (e.g. for wrapped spar construction, like a supra) that probably have vertical sides (but possibly at different specified angles).
* eliptical versions of above, or other shapes that constrain the bending radius of the spar-wrap to some structural parameter 
* placing a round CF tube (or rod) on the LE to protect it from dings. 
* Orienting a square-section timber rod on the LE at an orientation that minimises the sanding required to cut it down to the airfoil shape. i.e. cut an open beak shape into the LE.
* TE webbing (or control surfaces LE webbing) for basic hinged surfaces
* rectangular-section strips (usually non-tapered) normal to the wing surface, e.g. pulltruded cf square-rods or flat ribbons, timber strips, etc. that brace the skin from buckling. 

The last one is a bit funky. Do we say "the top surface of the spar has this shape, we want it to follow the top|bottom surface with these positions at the left/right ends, and be as close to the surface as possible without extending beyond it"? knowing how deep to cut the slot will require interpolating the Panel shape. e.g. if the spar is swept at a different angle to the wing, and the panel has different airfoils at each end, then the spar could need to pass under hills and valleys without sitting proud of the surface. I don't know what the right geometric strategy is or how to implement it. Maybe it's simpler to let the user specify the locations and complain/warn about proud surfaces?

While looking at this, maybe we should also consider hinges, which might be bastardised "full height section spars" or maybe a variant on that which resolves angles for wipers etc, or Fowler flap mechanisms, etc? Not trying to say that's in-scope for Spars, but if we approach it right may end up as more of the same. LE slots too.

Should the current LE/TE cutoffs be re-implemented as Spars too?
